boards/v2/qemu/qemu_malta/Kconfig.qemu_malta

```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2020 Antony Pavlov <antonynpavlov@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_QEMU_MALTA
+	select SOC_QEMU_MALTA
+	select QEMU_TARGET
+	select BIG_ENDIAN if "$(BOARD_IDENTIFIER)" = "/qemu_malta/be"
```

Comment:
Should move to `Kconfig` instead of `Kconfig.<board>`.

Also, `if "$(BOARD_IDENTIFIER)" = "/qemu_malta/be"` may be swapped to `if BOARD_QEMU_MALTA_BE` instead.
That will have the benefit that if board / soc / variant is ever renamed then it will result in Kconfig compliance failure which does not happen when comparing against a string.

Reply:
Did you mean `BOARD_QEMU_MALTA_QEMU_MALTA_BE`?

Reply:
actually this seems really broken, configure a project: `CONFIG_BOARD_QEMU_MALTA_QEMU_MALTA_BE` is set
Use menuconfig to edit and save, `CONFIG_BOARD_QEMU_MALTA_QEMU_MALTA_BE` is no longer present and `CONFIG_BOARD_QEMU_MALTA__BE` is now present

Reply:
> Did you mean `BOARD_QEMU_MALTA_QEMU_MALTA_BE`?

I did 👍

Reply:
> actually this seems really broken,

because you're missing this commit: https://github.com/zephyrproject-rtos/zephyr/commit/0691acac22de6cb4eb16652edb9433bcf096d653

Please rebase.

Reply:
commit 739bc5914888de9100ebe883ce9d68a21a14d2e3 (HEAD -> hwmv2-mips)
Author: Jamie McCrae <jamie.mccrae@nordicsemi.no>
Date:   Mon Jan 15 08:55:47 2024 +0000

    boards: mips: qemu_malta: Convert to v2
    
    Converts the board to hwmv2
    
    Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

commit 750ec32d9c4fd65f8288fd477055432d45c9bd18
Author: Jamie McCrae <jamie.mccrae@nordicsemi.no>
Date:   Mon Jan 15 08:50:38 2024 +0000

    soc: qemu_malta: Port to HWMv2
    
    Ports the qemu-malta SoC configuration to hardware model version 2
    
    Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

commit c8a6ce58f03f9aa9f6678b7d974be55b912be237 (origin/collab-hwm, collab-hwm)
Author: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
Date:   Wed Jan 10 13:13:37 2024 +0100

    cmake: fail when board identifier is applied in legacy hw model
    
    Board is looked up without identifier string, which means a legacy
    board may actually be found but later fails when board identifier is
    included.
    
    Thus test for HWMv1 and fail CMake is board identifier is provided for
    board using HWMv1.
    
    Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

commit 0691acac22de6cb4eb16652edb9433bcf096d653
Author: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
Date:   Wed Jan 10 12:24:30 2024 +0100

    cmake: cache BOARD_IDENTIFIER to preserve it between CMake invocations
    
    A CMake rerun would not extract or determine the BOARD_IDENTIFIER
    correctly because `list_boards.py` is only invoked for first CMake
    invocation.
    
    Therefore cache the BOARD_IDENTIFIER so that it's preserved between
    CMake re-runs.
    
    Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>


I have that commit

Reply:
First run: `-- Board: qemu_malta, variant: qemu_malta/be`
Run after menuconfig: `-- Board: qemu_malta, variant: variant: be`

Reply:
except you haven't pushed 739bc5914888de9100ebe883ce9d68a21a14d2e3.
At the time of this comment, the latest SHA of this PR is: ae4dd2d3d087b824fffbf8558173314a78242727, which doesn't contain 0691acac22de6cb4eb16652edb9433bcf096d653 commit.

Please push you local rebase so that I can verify if the issue can be reproduced.

Reply:
pushed

Reply:
Issue found and fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/67719

Reply:
Fix merged, please rebase.

Reply:
Applied

Reply:
comment here: https://github.com/zephyrproject-rtos/zephyr/pull/67592#discussion_r1454948729 still not addressed.